### PR TITLE
Releng/docs

### PR DESCRIPTION
**Pull request by jckautzmann**
Document the data layer integration with the core components

### CODE REVIEW

File: DATA_LAYER_INTEGRATION.md [hunk 1]
```diff
@@ -0,0 +1,280 @@
...
+The Core Components provide an out-of-the-box integration with the [Adobe Client Data Layer](https://github.com/adobe/adobe-client-data-layer), which for convenience is called data layer in this page.
+
+## Enabling the Data Layer
+
+The data layer is disabled by default. 
+
+To enable the data layer for your site:
+1. Create the following structure below the `/conf` node:
+    `/conf/<my-site>/sling:configs/com.adobe.cq.wcm.core.components.internal.DataLayerConfig`
+1. Add the `enabled` property and set it to `true`.
```
Comment (bpauli) [inline DATA_LAYER_INTEGRATION.md:28]:
```suggestion
1. Add the `enabled` boolean property and set it to `true`.
```

File: DATA_LAYER_INTEGRATION.md [hunk 2]
```diff
@@ -0,0 +1,280 @@
...
+The Core Components provide an out-of-the-box integration with the [Adobe Client Data Layer](https://github.com/adobe/adobe-client-data-layer), which for convenience is called data layer in this page.
+
+## Enabling the Data Layer
+
+The data layer is disabled by default. 
+
+To enable the data layer for your site:
+1. Create the following structure below the `/conf` node:
+    `/conf/<my-site>/sling:configs/com.adobe.cq.wcm.core.components.internal.DataLayerConfig`
+1. Add the `enabled` boolean property and set it to `true`.
```
Comment (vladbailescu) [inline DATA_LAYER_INTEGRATION.md:28]: There's an extra step needed:
- Add a `sling:configRef` to the root page `jcr:content` poiting the `/conf/<my-site>`

File: DATA_LAYER_INTEGRATION.md [hunk 3]
```diff
@@ -0,0 +1,280 @@
...
+```
+id: {
+    @type
+    repo:modifyDate
+    dc:title
+    dc:description
+    xdm:text
+    xdm:linkURL
+    parentId
+    image
```
Comment (vladbailescu) [inline DATA_LAYER_INTEGRATION.md:218]: I would add a comment here as well: ` // asset detail` to make it clear this is extra to the general component properties.

File: DATA_LAYER_INTEGRATION.md (hunk 2 above)
Comment (jckautzmann) [reply DATA_LAYER_INTEGRATION.md:28]: Addressed in https://github.com/adobe/aem-core-wcm-components/pull/1026/commits/3aef3ebdf90b8c4c43161e1cb2ebfca204c5edd1

File: DATA_LAYER_INTEGRATION.md (hunk 3 above)
Comment (jckautzmann) [reply DATA_LAYER_INTEGRATION.md]: Addressed in https://github.com/adobe/aem-core-wcm-components/pull/1026/commits/3aef3ebdf90b8c4c43161e1cb2ebfca204c5edd1

**Pull request merged**